wild/src/main.rs

```diff
@@ -1,4 +1,91 @@
+use anyhow::anyhow;
+use libc::fork;
+use std::env::args;
+use std::ffi::CString;
+use std::fs::File;
+use std::io::Error;
+use std::io::Read;
+use std::io::Write;
+
 fn main() -> wild_lib::error::Result {
-    let linker = wild_lib::Linker::from_env()?;
-    linker.run()
+    // skip the program name
+    let mut args: Vec<String> = args().skip(1).collect();
+
+    // See if there is a better way to do this, more canonical
+    let mut fork_subprocess = false;
+
+    // TODO decide if we want some short-form equivalent of "--fork"
+    args.retain(|a| {
+        if a == "--fork" {
+            fork_subprocess = true;
+            false
+        } else {
+            true
+        }
+    });
+
+    match fork_subprocess {
+        true => match make_named_pipe() {
+            Ok(path) => {
+                unsafe {
+                    match fork() {
+                        0 => {
+                            // Success in the parent
+                            // Wait for child to exit or pipe to be closed
+                            let mut f = File::open(path)?;
+                            let mut response = [0u8; 4];
+                            let count = f.read(&mut response)?;
+                            if count != 4 {
+                                return Err(anyhow!(
+                                    "Error retrieving exit status from child process"
+                                ));
+                            }
+                            let child_exit_status = i32::from_ne_bytes(response);
+                            std::process::exit(child_exit_status);
+                        }
+                        -1 => {
+                            // Failure in the parent
+                            // Create a linker with remaining args and run it
+                            wild_lib::Linker::from_args(args.into_iter())?.run()
+                        }
+                        _ => {
+                            // Success in the child
+                            // Create a linker with remaining args and run it
+                            wild_lib::Linker::from_args(args.into_iter())?.run()?;
+
+                            // inform parent that we are done!
+                            let mut f = File::open(path)?;
```

Comment:
To have child re-use an already open FD after fork, I think we'd need to make it a global, static variable...

Reply:
I'm pretty sure it should be possible to just have a local variable. Both parent and child process inherit the same stack

Reply:
I think you're right. If you do the call and have the variable before fork()